[WordPress → Jetpack app branding] Update mobile setup task

SiobhyB opened the pull request:
_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

The main focus of this PR involves updating the mobile set up task in Calypso's _My Home_ section to mention the Jetpack, rather than the WordPress, mobile app. 

<details>
 <summary>Code notes ⤵️</summary>

* The `jetpack/app-branding` feature flag (added in https://github.com/Automattic/wp-calypso/pull/66814) is used to only display the new Jetpack app mentions when the flag is set to true (currently only for wpcalypso and development builds).
* The animated icon added to the app banners in https://github.com/Automattic/wp-calypso/pull/67013 has been separated into its own component, to make it easier to re-use in other places in Calypso.

</details>


#### Testing Instructions

* Verify that the mobile setup task appears as expected when navigating to the _My Home_ page in Calypso.

It'd also be useful to verify the banner works as expected with the following:

* Different browsers.
* Different Calypso colour schemes.
* Different locales, including RTL.
* Note, the mobile setup task doesn't appear on mobile devices, so mobile testing isn't required though it'd still be useful to verify its appearance for smaller desktop screens.

#### Screenshots

<details>
 <summary>Design vs. Actual ⤵️</summary>
 
| Design  | Actual |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/193865317-61400cc6-1e54-44b6-a034-ebb7caa7f4bf.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194390382-57a1bca8-8793-4953-80b3-fe72da1d3073.png" width="100%"> |

</details>

<details>
 <summary>Before vs. After ⤵️</summary>

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/193865120-37900238-2654-4caf-90e5-e8fab5f7a600.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194390382-57a1bca8-8793-4953-80b3-fe72da1d3073.png" width="100%"> |

</details>



#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

SiobhyB committed:
Make it easier to implement animated icons

By separating the animated icon component from 'app-banner/index.jsx' to its own 'animated-icon/index.jsx' file, it will be easier to re-use the component's code and to implement animated icons elsewhere in Calypo's codebase.

SiobhyB committed:
Add Jetpack-specific details to app setup step

With this commit, a Jetpack-specific title, subtitle, icon, and description is added to the apps setup step when the 'jetpack/app-branding' feature flag is set to true.

SiobhyB committed:
Enable RTL version of animated icon

SiobhyB committed:
Styling: Use Jetpack green for button

SiobhyB committed:
Styling: Set height and width dimensions on icon

SiobhyB committed:
Use different paddings for when a task has an icon

SiobhyB commented on client/my-sites/customer-home/cards/tasks/site-setup-list/style.scss:
```diff
@@ -193,10 +193,22 @@
 
+	.site-setup-list__task-text.has-icon {
+		.task__description {
+			margin-top: -22px;
```
I'm not a fan of using negative margins, but I couldn't think of a better way to _only_ target the layout when there's an icon, without ending up doing a larger refactor (which didn't feel necessary). Interested to hear other thoughts on this.

dcalhoun replied:
Agreed, leveraging negative margin is not ideal. However, I also agree a larger refactor is not worthwhile at this time. 

My proposal would be to modify the design to embrace the existing gap used for task card elements, which means every task card would have the same gap balance. @shaunandrews WDYT? 

| Before | After | After (Inspection) |
| - | - | - |
| <img width="994" alt="Screen Shot 2022-10-05 at 10 09 50" src="https://user-images.githubusercontent.com/438664/194095742-9dda84ef-ddb2-47aa-b536-fda6b4c050c7.png"> | <img width="997" alt="balanced-gap" src="https://user-images.githubusercontent.com/438664/194095342-1f475615-9eaf-4ef3-8207-8f4266f0c299.png"> | <img width="999" alt="balanced-gap-inspect" src="https://user-images.githubusercontent.com/438664/194095336-45513e05-1eed-402d-8f25-d9828e8a7ae5.png"> | 

To accomplishing this, we would need to move the icon out of the header so that it is a sibling of the other gap elements. 

<details><summary>Relocated Element</summary>

<img width="639" alt="Screen Shot 2022-10-05 at 10 05 05" src="https://user-images.githubusercontent.com/438664/194095343-811dcba8-3913-40fd-a16d-d564aa243909.png">

</details>

shaunandrews replied:
It's fine to use the existing spacing.

SiobhyB replied:
Thanks both! I've gone ahead to revert back to the default spacing in https://github.com/Automattic/wp-calypso/pull/68511/commits/db8e3486f458a92152e6a1e901ba9c2b1e82cd53:

<img width="1103" alt="Screenshot 2022-10-06 at 18 10 43" src="https://user-images.githubusercontent.com/2998162/194376257-6e093c2b-3d65-4729-9a09-1e67e427dc19.png">

dcalhoun commented on client/components/animated-icon/index.jsx:
```diff
@@ -0,0 +1,33 @@
+	}, [ icon ] );
+
+	return (
+		<div ref={ iconEl } className={ classNames( 'app-animated-icon__container', className ) }></div>
```
Now that this its own component, I think it makes sense to have this be a top-level component class name. We could move the associated styles to `client/components/animated-icon/styles.scss`. WDYT? 
```suggestion
		<div ref={ iconEl } className={ classNames( 'animated-icon', className ) }></div>
```

SiobhyB replied:
I've gone ahead to commit the suggestion, thanks! 

> We could move the associated styles to client/components/animated-icon/styles.scss. WDYT?

I don't believe there are currently any generic styles that all instances of this component would share. The existing instances have heights and widths defined in CSS, but these vary.

Do you think there's anything generic that could be added to `animated-icon/styles.scss`? Or should we possibly remove the static `animated-icon` class name altogether?

dcalhoun replied:
Aha. I did not realize no styles are attached to the `animated-icon` selector. My initial thought was that we remove it if it is unused. However, it may provide some debugging value when viewing the source in a browser's developer tools. I'm fine with either retaining or removing the selector.

dcalhoun commented on client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:
```diff
@@ -133,6 +154,7 @@ export const getTask = (
 					actionDispatchArgs: [ siteId, task.id ],
 				} ),
 				isSkippable: true,
+				...jetpackAppBanner,
```
One thought: should we try to refactor this to colocate the mutation/conditional? 

This object spread conditionally mutates many of the existing object key values here. It is difficult to understand that without jumping back up to the `jetpackAppBanner` definition ~100 lines higher in the file. We could potentially (A) relocate the `jetpackAppBanner` assignment within this case statement or (B) refactor to conditionally spread without mutating. 

```js
// (A) Relocate `jetpackAppBanner` assignment
const jetpackAppBanner = displayJetpackAppBranding && { ...same as before... }
taskData = {
  ...same as before...
};

// (B) Avoid mutating
const wpAppBanner = { ... };
const jetpackAppBanner = { ... };

taskData = {
  ...( displayJetpackAppBranding ? jetpackAppBanner : wpAppBanner ),
  timing: 3,
  actionText: isBlogger
    ? translate( 'Download the app' )
    : translate( 'Download mobile app' ),
  actionUrl: '/me/get-apps',
  ...( ! task.isCompleted && {
    actionDispatch: requestSiteChecklistTaskUpdate,
    actionDispatchArgs: [ siteId, task.id ],
  } ),
  isSkippable: true,
};
```

SiobhyB replied:
I appreciate the feedback, I can see how the code was a bit confusing and like your approaches for making it clearer. I've gone with the second option, to avoid mutating: https://github.com/Automattic/wp-calypso/pull/68511/commits/7badcdaca1aa03eb137da7a1ec3f524c4b3e5723

dcalhoun commented on client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:
```diff
@@ -188,6 +189,7 @@ const SiteSetupList = ( {
 				userEmail,
 				isBlogger,
 				isFSEActive,
+				isRtl,
```
Should we import `useRtl` within `client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js` or is there a specific reason to pass this value here?

SiobhyB replied:
I get the `Invalid hook call. Hooks can only be called inside the body of a function component` error when attempting to use `useRtl` directly within the `getTask` function, so had assumed it was necessary to pass it down from the component. Maybe I'm missing another approach here?

dcalhoun replied:
Ah, yes. That makes total sense. Please discard my original thought.

dcalhoun commented on client/my-sites/customer-home/cards/tasks/site-setup-list/style.scss:
```diff
@@ -193,10 +193,22 @@
+	.site-setup-list__task-icon {
+		width: 67px;
+		height: 35px;
+		margin-bottom: 25px;
```
If we decide to refactor to rely upon the existing gap balance, this margin would likely need to be removed.

SiobhyB replied:
Addressed in https://github.com/Automattic/wp-calypso/commit/db8e3486f458a92152e6a1e901ba9c2b1e82cd53, following the discussion [here](https://github.com/Automattic/wp-calypso/pull/68511#discussion_r987068315).

dcalhoun reviewed (commented):
This looks good so far. I left a few comments for consideration, but nothing I consider a blocker.

SiobhyB committed:
Revert to using task card's default padding

This commit follows the feedback here: https://github.com/Automattic/wp-calypso/pull/68511#discussion_r988044987

Instead of attempting to define custom padding in cases where there is an icon, we are going to be using the card's default gap/padding.

SiobhyB committed:
Tweak top-level component's class name for clarity

Co-authored-by: David Calhoun <438664+dcalhoun@users.noreply.github.com>

SiobhyB committed:
Refactor code for clarity

Following feedback here, the conditional code determining whether to display copy specific to the WP or Jetpack app has been refactored for clarity: https://github.com/Automattic/wp-calypso/pull/68511#discussion_r988019659

SiobhyB committed:
Merge branch 'update/mobile-set-up-task' of github.com:Automattic/wp-calypso into update/mobile-set-up-task

SiobhyB committed:
Address formatting errors

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7573134

Thank you @SiobhyB for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7573134).
